### DISCUSSION TITLE
Anaconda

### DISCUSSION THREAD

**Discussion by amotl** [crate/crate-clients-tools#198] [Ideas]
## About
We as Linux/macOS users often underestimate the large world-coverage of Windows installations. I think it could make a difference to provide first-class Conda packages.

- https://github.com/crate/crash/issues/165
- https://github.com/crate/roadmap/issues/31

By using automation, it is not rocket science at all: For [Wetterdienst](https://github.com/earthobservations/wetterdienst), we got friendly help from @Xylar when doing our first orientation flights on @conda-forge. We learned that most things will get staged through GitHub, with additional automation and help from the venerable @conda-bot.

## Thoughts
Is it something you could add to your "also doing a share of maintenance" agenda, @simonprickett and @kneth? In my opinion, it also relates to devex and devrel very much.

**Comment by amotl** [comment]
### pixi - powerful development environments on Windows, macOS and Linux

> [pixi](https://github.com/prefix-dev/pixi) is a fast software package manager built on top of the existing conda ecosystem. Spins up development environments quickly on Windows, macOS and Linux.
>
> Automatic lockfiles produce reproducible environments across operating systems (without Docker!).
>
> pixi supports Python, R, C/C++, Rust, Ruby, and many other languages.

- https://prefix.dev/
- https://github.com/prefix-dev/pixi

**Comment by xylar** [comment]
Happy to help on the conda-forge side if you need.

**Reply by amotl** [reply]
Hi Xylar,

so cool. Willing to help, as always, you are great!

We will humbly let you know when we finally get around publishing our Python artefacts (crate-python, sqlalchemy-cratedb, cratedb-sqlparse, cratedb-toolkit, mlflow-cratedb, commons-codec) to the conda forges, and maybe not just the Python ones. As we can see, Conda is also providing packages for traditional database servers like PostgreSQL or MySQL, and probably also many others?

Do you know about any blueprints how to publish a Java-based program like [CrateDB](https://github.com/crate/crate) itself to Conda? Is it even possible, or advisable?

With kind regards,
Andreas.

**Reply by xylar** [reply]
@amotl, I have absolutely no experience with java-based packages on conda-forge.  I see that openjdk is available:
https://github.com/conda-forge/openjdk-feedstock

I also see that bunches of packages use it:
https://github.com/search?q=path%3A**%2Frecipe%2Fmeta.yaml+org%3Aconda-forge+openjdk&type=code&ref=advsearch

It seems like the easiest starting point would be to find one (presumably one that's relatively popular, based on number of downloads, but not too complicated).  You could ping `(at)conda-forge/help-java` (mangled because I don't actually want to ping them here...) to possibly get some assistance but I don't have any idea how responsive they might be.

**Reply by amotl** [reply]
Sweet, backtracking openjdk-feedstock sounds like a good plan, thanks. If I will run into any problems with the Python packages then, I may ping you, and spare you about the Java stuff ;]. Thanks a stack for sharing relevant pointers where to reach out in this case. 🙇
